src/frontends/pytorch/src/op/max_poolnd.cpp

```diff
@@ -63,19 +89,62 @@ OutputVector translate_max_poolnd(const NodeContext& context) {
                                                                 PadType::EXPLICIT,
                                                                 element::i64,
                                                                 2));
-    if (context.get_output_size() == 2) {
-        auto out1 = res->output(0);
-        auto out2 = res->output(1);
-        return {std::move(out1), std::move(out2)};
+    if (is_static) {
+        if (no_batch_dim) {
+            if (context.get_output_size() == 2) {
+                auto out1 = res->output(0);
+                auto out2 = res->output(1);
+                out1 = context.mark_node(std::make_shared<v0::Squeeze>(out1, const_0));
+                out2 = context.mark_node(std::make_shared<v0::Squeeze>(out2, const_0));
+                return {std::move(out1), std::move(out2)};
+            } else {
+                res = context.mark_node(std::make_shared<v0::Squeeze>(res, const_0));
+                return {res};
+            }
+        }
```

Comment:
You don't have else for `if (no_batch_dim)`, because of that your code can't be built.